rpcs3/rpcs3qt/welcome_dialog.cpp

```diff
@@ -0,0 +1,93 @@
+#include "welcome_dialog.h"
+
+#include "Emu/System.h"
+
+#include <QPushButton>
+#include <QCheckBox>
+#include <QLabel>
+#include <QVBoxLayout>
+#include <QHBoxLayout>
+
+welcome_dialog::welcome_dialog(QWidget* parent) : QDialog(parent)
+{
+	QPushButton* okay = new QPushButton(tr("Okay"));
+	okay->setEnabled(false);
+
+	QCheckBox* do_not_show = new QCheckBox(tr("Do not show again"));
+	QCheckBox* i_have_read = new QCheckBox(tr("I have read the quickstart guide (required)"));
+
+	connect(i_have_read, &QCheckBox::stateChanged, [=](int state)
+	{
+		okay->setEnabled(state != 0);
+	});
+
+	connect(do_not_show, &QCheckBox::stateChanged, [=](int state)
+	{
+		std::string val = state == 0 ? "true" : "false";
+		g_cfg.misc.show_welcome_screen.from_string(val);
+	});
+
+	connect(okay, &QPushButton::pressed, this, &QDialog::accept);
+
+	QIcon rpcs3_icon = QIcon(":/rpcs3.ico");
+	QLabel* icon = new QLabel(this);
+	icon->setPixmap(rpcs3_icon.pixmap(96, 96));
+
+	QLabel* caption = new QLabel(tr(
+		"<h1>Welcome to RPCS3</h1>"
+		"<h2>An open-source PlayStation 3 emulator for Windows and Linux funded by <a href =\"https://www.patreon.com/Nekotekina\">Patreon</a>!</h2><br>"
+		"To get started, you need to install the PlayStation 3 firmware. "
+		"Check out our <a href=\"https://rpcs3.net/quickstart\">quickstart guide</a> for further information. "
+		"If you have any questions, please have a look at our <a href =\"https://rpcs3.net/faq\">FAQ</a>. "
+		"Otherwise, further discussion and support can be found at our <a href =\"http://www.emunewz.net/forum/forumdisplay.php?fid=172\">Forum</a> "
+		"or <a href =\"https://discord.me/RPCS3\">Discord</a>.<br>"
+	));
+
+	caption->setFixedWidth(450);
+	caption->setWordWrap(true);
+	caption->setOpenExternalLinks(true);
+
+	QLabel* piracy = new QLabel(tr(
+		"<font color='red'><u><b>RPCS3 does NOT support piracy. You must dump your own games.</b></u></font>"
+	));
+	piracy->setWordWrap(true);
+	piracy->setTextFormat(Qt::RichText);
+
+	icon->setAlignment(Qt::AlignLeft);
+	caption->setAlignment(Qt::AlignLeft);
```

Comment:
maybe put those to their object's other function calls. They come out of nothing